app/components/MediaList/MediaList.jsx

```diff
@@ -0,0 +1,144 @@
+import * as Constants from 'lib/constants'
+import {connectRedux} from 'lib/redux'
+import DocumentGridList from 'components/DocumentGridList/DocumentGridList'
+import DocumentTableList from 'containers/DocumentTableList/DocumentTableList'
+import DropArea from 'components/DropArea/DropArea'
+import FieldMediaItem from 'components/FieldMedia/FieldMediaItem'
+import MediaGridCard from 'containers/MediaGridCard/MediaGridCard'
+import MediaListController from 'components/MediaListController/MediaListController'
+import proptypes from 'prop-types'
+import React from 'react'
+
+const MEDIA_TABLE_FIELDS = ['url', 'fileName', 'mimeType', 'width', 'height']
+
+class MediaList extends React.Component {
+  static propTypes = {
+    /**
+     * The list of documents shown in the list.
+     */
+    documents: proptypes.array,
+
+    /**
+     * Whether the view is filtered to only show selected documents.
+     */
+    isFilteringSelection: proptypes.bool,
+
+    /**
+     * Whether any documents are selected.
+     */
+    hasSelection: proptypes.bool,
+
+    /**
+     * A callback to be used to obtain the base URL for the given page, as
+     * determined by the view.
+     */
+    onBuildBaseUrl: proptypes.func,
+
+    /**
+     * Callback to be fired when the selection changes.
+     */
+    onSelect: proptypes.func,
+
+    /**
+     * The order used to sort the documents by the `sort` field.
+     */
+    order: proptypes.oneOf(['asc', 'desc']),
+
+    /**
+     * The list of currently selected documents.
+     */
+    selectedDocuments: proptypes.object,
+
+    /**
+     * The name of the field currently being used to sort the documents.
+     */
+    sort: proptypes.string
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.renderFieldMediaItem = ({document}) => (
```

Comment:
I'm not sure I like the idea of having this in the constructor. I know that it's a single JSX node, but I think it creates a weird precedent. Would you mind moving this to a function?